languages/fsharp/exercises/concept/dates/DatesTests.fs

```diff
@@ -0,0 +1,134 @@
+module DatesTests
+
+open FsUnit.Xunit
+open Xunit
+open Xunit.Sdk
+open System
+open System.Globalization
+open System.Reflection
+open System.Threading
+
+open Dates
+
+[<AttributeUsage(AttributeTargets.Class ||| AttributeTargets.Method)>]
+type UseCultureAttribute(cultureName: string, uiCultureName: string) =
+    inherit BeforeAfterTestAttribute()
+
+    let culture = CultureInfo(cultureName, false)
+    let uiCulture = CultureInfo(uiCultureName, false)
+
+    let mutable originalCulture: CultureInfo = culture
+    let mutable originalUiCulture: CultureInfo = uiCulture
+
+    new(cultureName: string) = UseCultureAttribute(cultureName, cultureName)
+
+    override __.Before(_: MethodInfo) =
+        originalCulture <- Thread.CurrentThread.CurrentCulture
+        originalUiCulture <- Thread.CurrentThread.CurrentUICulture
+
+        Thread.CurrentThread.CurrentCulture <- culture
+        Thread.CurrentThread.CurrentUICulture <- uiCulture
+
+        CultureInfo.CurrentCulture.ClearCachedData()
+        CultureInfo.CurrentUICulture.ClearCachedData()
+
+    override __.After(_: MethodInfo) =
+        Thread.CurrentThread.CurrentCulture <- originalCulture
+        Thread.CurrentThread.CurrentUICulture <- originalUiCulture
+
+        CultureInfo.CurrentCulture.ClearCachedData()
+        CultureInfo.CurrentUICulture.ClearCachedData()
+
+[<UseCulture("en-US")>]
+type AppointmentTests() =
+
+    [<Fact>]
+    let ``Schedule date using only numbers``() =
+        schedule "7/25/2019 13:45:00" |> should equal (DateTime(2019, 07, 25, 13, 45, 0))
+
+    [<Fact>]
+    let ``Schedule date with textual month``() =
+        schedule "June 3, 2019 11:30:00" |> should equal (DateTime(2019, 6, 3, 11, 30, 0))
+
+    [<Fact>]
+    let ``Schedule date with textual month and weekday``() =
+        schedule "Thursday, December 5, 2019 09:00:00" |> should equal (DateTime(2019, 12, 5, 9, 0, 0))
+
+    [<Fact>]
+    let ``Has passed with appointment 1 year ago``() =
+        hasPassed (DateTime.Now.AddYears(-1).AddHours(2.0)) |> should equal true
+
+    [<Fact>]
+    let ``Has passed with appointment months ago``() = hasPassed (DateTime.Now.AddMonths(-8)) |> should equal true
+
+    [<Fact>]
+    let ``Has passed with appointment days ago``() = hasPassed (DateTime.Now.AddDays(-23.0)) |> should equal true
```

Comment:
Any reason you're including the `.0` here? (same question for multiple instances below)
```suggestion
    let ``Has passed with appointment days ago``() = hasPassed (DateTime.Now.AddDays(-23)) |> should equal true
```

Reply:
Yes, because weirdly enough the [AddDays method](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.adddays?view=netcore-3.1) only takes a `double` (there is no `int` overload).